### PR TITLE
Adding the QuantumBridge (PennyLane support)

### PR DESCRIPTION
**Pull request by CassNot**
## Context / Related Issues

This branch propose a QuantumBridge implementation (and fix a minor type issue in benchmark_slos_core). This is based on a code from @ben9871 !

The `QuantumBridge` provides a seamless interface between (PennyLane's or other) qubit-based quantum circuits and Merlin's photonic quantum processors. Examples are given in the documentation !

Module is a standalone no-parameter module converting a qubit based input into a specified encoding output. see documentation for more details

## Type of change
- [ ] Bug fix
- [x] New feature
- [x] Documentation update
- [ ] Refactor / Cleanup
- [ ] Performance improvement
- [ ] CI / Build / Tooling
- [ ] Breaking change (requires migration notes)

## Proposed changes

- a QuantumBridge implementation in bridge/quantum_bridge
- associated tests on CPU and GPU (one test is done with PennyLane if there is PennyLane in the venv)

## How to test / How to run

You can run the pytest on the files in tests/bridge

See documentation for example of usage.

## Documentation
- [x] User docs updated (Sphinx)
- [ ] Examples / notebooks updated
- [x] Docstrings updated

## Checklist
- [x] Code formatted (ruff format)
- [x] Lint passes (ruff)
- [x] Static typing passes (mypy) if applicable
- [x] Unit tests added/updated (pytest)
- [x] Tests pass locally (pytest)
- [x] Tests pass on GPU (pytest)
- [x] Test coverage not decreased significantly
- [ ] Docs build locally if affected (sphinx)
- [x] Dependencies updated (if needed) and pinned appropriately
- [x] I have added a clear PR title and description

### CODE REVIEW

**Review by v-espitalier — COMMENTED** [review]
Overall, the PR is well-structured and the integration of QuantumBridge into MerLin is clear. My comments focus on improving long-term reliability (e.g., device handling, memory efficiency) and simplifying redundant checks (e.g., imports, type conversions) to align with MerLin’s design patterns.

File: merlin/bridge/quantum_bridge.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        x -> PennyLane state ψ (complex) -> superposition over SLOS keys -> Merlin output.
+        """
+        psi = self._call_pl_state(x)  # (B,K) complex
+        # Unify dtype/device with the bridge/merlin side
+        target_complex = (
+            torch.complex64
+            if self.dtype in (torch.float32, torch.bfloat16)
+            else torch.complex128
+        )
+        target_device = self.device if self.device is not None else psi.device
```
[the remark this answers is not in the document]
Comment (jsenellart) [reply merlin/bridge/quantum_bridge.py]: removed black-magic - no automatic moving during forward, transition matrix is constructed on specified device or cpu - and then will move with .to method

File: merlin/bridge/quantum_bridge.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        psi = psi.to(dtype=target_complex, device=target_device)
+
+        if self.normalize:
+            psi = psi / (psi.norm(dim=1, keepdim=True) + 1e-20)
```
[the remark this answers is not in the document]
Comment (jsenellart) [reply merlin/bridge/quantum_bridge.py]: this is actually more general, we have this constants in a lot of merlin module, normalization should be defined once for all in a single file. will be done later

File: merlin/bridge/quantum_bridge.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+import torch
+import torch.nn as nn
+
+try:
+    # Requires the 'merlin' package exposing QuantumLayer
+    from merlin import QuantumLayer
+except Exception as e:  # pragma: no cover
+    raise ImportError(
+        "This bridge requires 'merlin' with QuantumLayer installed."
+    ) from e
```
[the remark this answers is not in the document]
Comment (jsenellart) [reply merlin/bridge/quantum_bridge.py]: done

**Review by ben9871 — APPROVED** [review]
I accept these changes, it is what we discussed previously
